integrations/messenger/src/misc/incoming-message.ts

```diff
@@ -13,46 +13,69 @@ export async function handleMessage(
     logger,
   }: { client: bp.Client; ctx: IntegrationContext<bp.configuration.Configuration>; logger: IntegrationLogger }
 ) {
-  if (message?.message?.text) {
-    logger.forBot().debug('Received text message from Messenger:', message.message.text)
-    const { conversation } = await client.getOrCreateConversation({
-      channel: 'channel',
-      tags: {
-        id: message.sender.id,
-        senderId: message.sender.id,
-        recipientId: message.recipient.id,
-      },
-    })
-
-    const { user } = await client.getOrCreateUser({
-      tags: {
-        id: message.sender.id,
-      },
-    })
-
-    // TODO: do this for profile_pic as well, as of 13 NOV 2023 the url "https://platform-lookaside.fbsbx.com/platform/profilepic?eai=<eai>&psid=<psid>&width=<width>&ext=<ext>&hash=<hash>" is not working
-    if (!user.name) {
-      try {
-        const messengerClient = getMessengerClient(ctx.configuration)
-        const profile = await messengerClient.getUserProfile(message.sender.id, { fields: ['id', 'name'] })
-        logger.forBot().debug('Fetched latest Messenger user profile: ', profile)
-
-        await client.updateUser({ ...user, name: profile.name })
-      } catch (error) {
-        logger.forBot().error('Error while fetching user profile from Messenger:', error)
-      }
-    }
+  const { sender, recipient, message: textMessage, postback } = message
+
+  if (!textMessage?.text && !postback?.payload) {
```

Comment:
**chore:** this piece of code is written in a way where:
1. your intention is not clear
2. you don't ask help from your compiler to catch mistakes; it would be very easy to introduce a regression later on

I would probably write something like this:

```ts
const { sender, recipient, message: textMessage, postback } = message

// Here you shouldn't declare these variables with empty string values. If typescript detects a path where these variables arent set, it will complain (that's what we want)
let text: string
let messageId: string

if (textMessage?.text) {
    text = textMessage.text
    messageId = textMessage.mid
    // no need to declare a logMessage variable for a single line log
    logger.forBot().debug(`Received text message from Messenger: ${textMessage.text}`)
} else if (postback?.payload) {
    // here using the postback if mutually exclusive with using the text message (you can't enter both branch, so it is easier to follow)
    text = postback.payload
    messageId = postback.mid
    logger.forBot().debug(`Received postback from Messenger: ${postback.title}`)
} else {
    return
}
```